src/app.py

```diff
@@ -3,103 +3,124 @@
+    Input('genre_f', 'value'),
+    Input('region_f', 'value'))
+def lineGenreCallback(genre, region):
+    return plotLineGenre(genre, region)
 
-app = Dash(__name__,  external_stylesheets=[dbc.themes.BOOTSTRAP])
 
-app.layout = dbc.Container([
-    #html.H1('Video Game Sales Dashboard'),
-    dbc.Row([
-        dbc.Col(
-        html.Div([
-        html.H3('Video Game Sales', style={'color':'#ffffff'}),
+@app.callback(
+    Output('publishers', 'srcDoc'),
+    Input('topn-slider', 'value'))
+def update_output(value):
+    return plotPublishers(value)
+
+
+# Chart drawing functions
+def lineGlobalFigure():
+    return  html.Iframe(
+                    id=f'line_global',
+                    style={'border-width': '0', 'width': '100%', 'height': '400px'},
+                    srcDoc=plotLineGlobal(genre='Sports'))
+
+
+def lineGenreFigure():
+    return  html.Iframe(
+                    id=f'line_genre',
+                    style={'border-width': '0', 'width': '100%', 'height': '400px'},
+                    srcDoc=plotLineGenre(genre='Sports', region='North America'))
+
+def publishersFigure():
+    return  html.Div([
+        html.Iframe(
+                    id=f'publishers',
+                    style={'border-width': '0', 'width': '100%', 'height': '400px'},
+                    srcDoc=plotPublishers(top_val='10')),
+         dcc.Slider(0, 20, 5,
+               value=10,
+               id='topn-slider')
+    ])
+
+app.title = "Video Game Sales"
+
+SIDEBAR_STYLE = {
+    "position": "fixed",
+    "top": 0,
+    "left": 0,
+    "bottom": 0,
+    "width": "16rem",
+    "padding": "2rem 1rem",
+    "background-color": "#f8f9fa",
+}
+
+# the styles for the main content position it to the right of the sidebar and
+# add some padding.
+CONTENT_STYLE = {
+    "margin-left": "18rem",
+    "margin-right": "2rem",
+    "padding": "2rem 1rem",
+}
+
+sidebar = html.Div(
+    [
+       html.H3('Video Game Sales'),
         html.Br(),
-        html.H5('Genre', style={'color':'#ffffff'}),
+        html.H5('Genre'),
         dcc.Dropdown(
             id='genre_f',
             value='Sports',
-            options=[{'label': i, 'value': i} for i in pivoted['Genre'].unique()]),
+            options=[{'label': i, 'value': i} for i in getGenres()]),
         html.Br(),
-        html.H5('Region', style={'color':'#ffffff'}),
+        html.H5('Region'),
         dcc.Dropdown(
             id='region_f',
             value='North America',
-            options=[{'label': i, 'value': i} for i in pivoted['Region'].unique()])
+            options=[{'label': i, 'value': i} for i in getRegions()])
+    ],
+    style=SIDEBAR_STYLE,
+)
+
+content = html.Div([
+                dbc.Row([
+                        dbc.Col([
+                            lineGlobalFigure()
+                        ]),
+                        dbc.Col([
+                            lineGenreFigure()
+                        ]),
+                ]),
+
+                dbc.Row([
```

Comment:
i like this . We dont modify any other part of the dashboard just the content and the dropdown and add the plotting functions